src/components/subnavigation/hooks/use-outside-click.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react';
+
+const useOutsideClick = (ref: any, callback: any) => {
+    const handleClick = (e: MouseEvent) => {
+        if (ref.current && !ref.current.contains(e.target)) {
+            callback();
+        }
+    };
+
+    useEffect(() => {
+        document.addEventListener('click', handleClick);
+
+        return () => {
+            document.removeEventListener('click', handleClick);
+        };
+    });
+};
+
+export default useOutsideClick;
```

Comment:
Love this hook! But should we have this `hooks` directory at a higher level (`src/hooks`) for reusability throughout the project?